src/test_disk.rs

```diff
@@ -1248,17 +1296,34 @@ pub fn check_all_disks(
                         }
                     }
                     false
-                }).and_then(|r| Some(r.clone()));
+                })
+                .and_then(|r| Some(r.clone()));
             debug!("thread {} scsi_info: {:?}", process::id(), scsi_info);
             debug!("thread {} device: {:?}", process::id(), device);
-            let mut s = StateMachine::new(device, scsi_info, false);
+            let mut s = StateMachine::new(device.clone(), scsi_info, false);
             s.setup_state_machine();
-            s.block_device.state = get_state(pool, &s.block_device)?;
+            // I think this is the problem.  This restores the previous state
```

Comment:
Once we have the information in the database in add_previous_devices(), is this block still needed?